dbt/include/athena/macros/materializations/models/helpers_iceberg.sql

```diff
@@ -101,6 +103,40 @@
     );
 {%- endmacro %}
 
+{% macro iceberg_merge(tmp_relation, target_relation, unique_key, statement_name="main") %}
+    {%- set dest_columns = adapter.get_columns_in_relation(target_relation) -%}
+    {%- set dest_cols_csv = dest_columns | map(attribute='quoted') | join(', ') -%}
+    {% if unique_key is sequence and unique_key is not string %}
+      {%- set unique_key_cols = unique_key -%}
+    {% else %}
+      {%- set unique_key_cols = [unique_key] -%}
+    {% endif %}
+    {%- set src_columns = [] -%}
+    {%- set update_columns = [] -%}
+    {%- for col in dest_columns -%}
+      {%- do src_columns.append('tmp_rel."' + col.name + '"') -%}
+      {% if col.name not in unique_key_cols %}
+        {%- do update_columns.append(col.name) -%}
+      {% endif %}
+    {%- endfor -%}
+    {%- set src_cols_csv = src_columns | join(', ') -%}
+
+    merge into {{ target_relation }} as tar_rel using {{ tmp_relation }} as tmp_rel
```

Comment:
nit: `tmp_rel ` could be called `src` or `source`, `tar_rel ` can be called `target`.